### ISSUE TITLE
Add support for LaTeX font selection commands

### ISSUE THREAD

**Issue by michal-h21** [michal-h21/luatex-harfbuzz-shaper#12]
Can we somehow use NFSS? Can we add support for `harfbuzz` to `fontspec`?

**Comment by deepakjois** [comment]
On Wed, Dec 16, 2015 at 1:40 PM, Michal Hoftich
notifications@github.com wrote:

> Can we somehow use NFSS? Can we add support for harfbuzz to fontspec?

I don’t know much a lot NFSS or fontspec. But my impression was that
shaping etc. happens in _luaotfload_. _luaotfload_ in turn uses a
bundled copy of ConTeXt’s fontloader. What would be nice is to have
harfbuzz as an option in luaotfload, in addition to ConTeXt's
fontloader.

**Comment by michal-h21** [comment]
Yes, inclusion in luatfload would be nice, but it probably depends on inclusion of libraries in TeX Live. With some simple font selection scheme we can test real LaTeX documents right now.

**Comment by davidcarlisle** [comment]
On 16 December 2015 at 09:02, Michal Hoftich notifications@github.com
wrote:

> Yes, inclusion in luatfload would be nice, but it probably depends on
> inclusion of libraries in TeX Live. With some simple font selection scheme
> we can test real LaTeX documents right now.

@josephwright and I could probably look at latex/NFSS but in my case I'd
probably need to build on cygwin as that's the texlive platform I usually
use. Will look later...

David

**Comment by michal-h21** [comment]
I've renamed and modified files eu2enc.def and eu2lmr.fd, provided by
fontspec, as EU3 font encoding and it seems to work nicely. Can
`\DeclareFontShape` comand be used without `.fd` files?

> @josephwright and I could probably look at latex/NFSS but in my case I'd
> probably need to build on cygwin as that's the texlive platform I usually
> use. Will look later...

I am quite interested in building on Windows, I've tried it, but wasn't
successful.

Michal

**Comment by josephwright** [comment]
I've pushed the team already to say we shouldn't have more than one Unicode encoding (it's a font loading variation, not an encoding one). A third 'encoding' makes this more urgent: I'll push the rest of the team again to address this properly. (It's quite doable, just needs a decision.)

**Comment by davidcarlisle** [comment]
On 16 December 2015 at 09:35, Michal Hoftich notifications@github.com
wrote:

> I've renamed and modified files eu2enc.def and eu2lmr.fd, provided by
> fontspec, as EU3 font encoding and it seems to work nicely.

At this stage just whatever is easiest although there is an outstanding
issue on fontspec
to use the same encoding (called say UC for unicode) with luatex and xetex
so same
could be used here, the different encodings are not really different
encodings just
used as a hook for minor differences in font loading syntax, but having
different encodings
complicates other things.

> Can
> `\DeclareFontShape` comand be used without `.fd` files?

Yes, fontspec only needs fd files normally for fonts like latin modern that
exist in a range of
design sizes. In the normal system font case where just using one scaled
font, it does not
need them.

> > @josephwright and I could probably look at latex/NFSS but in my case I'd
> > probably need to build on cygwin as that's the texlive platform I usually
> > use. Will look later...
> 
> I am quite interested in building on Windows, I've tried it, but wasn't
> successful.
> 
> Michal
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/michal-h21/luatex-harfbuzz-shaper/issues/12#issuecomment-165046806
> .

## 

http://dpcarlisle.blogspot.com/

**Comment by michal-h21** [comment]
> At this stage just whatever is easiest although there is an outstanding
> issue on fontspec
> to use the same encoding (called say UC for unicode) with luatex and xetex
> so same
> could be used here, the different encodings are not really different
> encodings just
> used as a hook for minor differences in font loading syntax, but having
> different encodings
> complicates other things.
> 
> Ah, fine, I thought it would be better to use own encoding, but EU2 works
> as well.
> 
> > Can
> > `\DeclareFontShape` comand be used without `.fd` files?
> 
> Yes, fontspec only needs fd files normally for fonts like latin modern that
> exist in a range of
> design sizes. In the normal system font case where just using one scaled
> font, it does not
> need them.
> 
> Great, it makes things quite easy, I think :)

**Comment by michal-h21** [comment]
I've added simple font selection: f7a2cc07f55197974eb629523dcd1a3adab2b306
